flink-filesystems/flink-s3-fs-base/src/main/java/org/apache/flink/fs/s3/common/FlinkS3FileSystem.java

```diff
@@ -101,13 +237,154 @@ public FlinkS3FileSystem(
         this.s3AccessHelper = s3UploadHelper;
         this.uploadThreadPool = Executors.newCachedThreadPool();
 
-        Preconditions.checkArgument(s3uploadPartSize >= S3_MULTIPART_MIN_PART_SIZE);
+        checkArgument(s3uploadPartSize >= S3_MULTIPART_MIN_PART_SIZE);
         this.s3uploadPartSize = s3uploadPartSize;
         this.maxConcurrentUploadsPerStream = maxConcurrentUploadsPerStream;
+        LOG.info("Created Flink S3 FS, s5Cmd configuration: {}", s5CmdConfiguration);
     }
 
     // ------------------------------------------------------------------------
 
+    @Override
+    public boolean canCopyPaths(Path source, Path destination) {
+        return canCopyPaths();
+    }
+
+    private boolean canCopyPaths() {
+        return s5CmdConfiguration != null;
+    }
```

Comment:
1. Should we also check that one is remote and one is local? (IIRC, this is s5cmd requirement)
2. Use public method everywhere and inline the private one?